From 16c41745c7b92a243d0874f534c1655196c64b74 Mon Sep 17 00:00:00 2001
From: Phil Blundell <philb@gnu.org>
Date: Wed, 24 Nov 2010 11:49:53 -0800
Subject: [PATCH] econet: fix CVE-2010-3850

Add missing check for capable(CAP_NET_ADMIN) in SIOCSIFADDR operation.

Signed-off-by: Phil Blundell <philb@gnu.org>
Signed-off-by: David S. Miller <davem@davemloft.net>
---
 net/econet/af_econet.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

Index: linux-2.6.36-gentoo-r3/net/econet/af_econet.c
===================================================================
--- linux-2.6.36-gentoo-r3.orig/net/econet/af_econet.c
+++ linux-2.6.36-gentoo-r3/net/econet/af_econet.c
@@ -671,6 +671,9 @@ static int ec_dev_ioctl(struct socket *s
 	err = 0;
 	switch (cmd) {
 	case SIOCSIFADDR:
+		if (!capable(CAP_NET_ADMIN))
+			return -EPERM;
+
 		edev = dev->ec_ptr;
 		if (edev == NULL) {
 			/* Magic up a new one. */
